### PR TITLE
AzureMonitor: Show error message when subscriptions request fails in ConfigEditor

### LINKED ISSUE #34426: AzureMonitor: show error message in config editor if cannot fetch list of subscriptions
Currently, if entered credentials are invalid and Azure Monitor config editor is unable to fetch list of subscriptions, then the subscriptions dropdown list just silently doesn't update.

We should provide visual indication that config editor was unable to fetch the list of subscriptions.
Also make sure that default subscription choice being reset in case of failure.

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Shows error message if the request to get subscriptions fail.

**Which issue(s) this PR fixes**:

Fixes #34426

**Special notes for your reviewer**:

### CODE REVIEW

**Review by sarahzinger — APPROVED** [review]
Seems like a good idea to me! If it's easy it might be nice to add tests for this sort of thing but no pressure if it feels more effort than it's worth, this seems relatively straightforward.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx [hunk 1]
```diff
@@ -102,6 +125,13 @@ export class ConfigEditor extends PureComponent<Props, State> {
             onResetOptionKey={this.resetSecureKey}
           />
         )}
+
+        {error && (
```
Comment (vickyyyyyyy) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx:129]: how does this work with the `DataSourceSettingsPage`?
https://github.com/grafana/grafana/blob/main/public/app/features/datasources/settings/DataSourceSettingsPage.tsx#L209

Does this plugin use the above so this error ends up being rendered before the `DataSourceSettingsPage` or is this not used at all?

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx [hunk 2]
```diff
@@ -102,6 +125,13 @@ export class ConfigEditor extends PureComponent<Props, State> {
...
+          <Alert severity="error" title={error.title}>
```
Comment (vickyyyyyyy) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx:130]: would be good to have a test for this

**Review by vickyyyyyyy — APPROVED** [review]
LGTM, just left a question and suggestion

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx (hunk 1 above)
Comment (vickyyyyyyy) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx:129]: gotcha 👍

**Pull request merged**